packages/fluentui/react-northstar/src/components/Tree/Tree.tsx

```diff
@@ -96,108 +104,101 @@ export interface TreeItemForRenderProps {
-    selectable: PropTypes.bool,
-    items: customPropTypes.collectionObjectShorthand,
-    onActiveItemIdsChange: PropTypes.func,
-    onSelectedItemIdsChange: PropTypes.func,
-    renderItemTitle: PropTypes.func,
-    renderedItems: PropTypes.func,
-  };
-
-  static defaultProps = {
-    as: 'div',
-    accessibility: treeBehavior as Accessibility,
-  };
-
-  static autoControlledProps = ['activeItemIds', 'selectedItemIds'];
-
-  static Item: typeof TreeItem = TreeItem;
-  static Title: typeof TreeTitle = TreeTitle;
-
-  static getAutoControlledStateFromProps(nextProps: TreeProps, prevState: TreeState) {
-    const { items, selectable } = nextProps;
-    let { activeItemIds, selectedItemIds } = nextProps;
-
-    if (selectable && items && !selectedItemIds) {
-      if (!selectedItemIds && items) {
-        selectedItemIds = prevState.selectedItemIds;
-
-        const iterateItems = (items, selectedItems = selectedItemIds) => {
-          _.forEach(items, item => {
-            if (item['selected'] && selectedItemIds.indexOf(item['id']) === -1) {
-              selectedItems.push(item['id']);
-            }
-            if (item['items']) {
-              return iterateItems(item['items']);
-            }
-          });
-        };
-
-        iterateItems(items);
-      }
-    }
-
-    if (!activeItemIds && items) {
-      activeItemIds = prevState.activeItemIds;
-
-      const expandedItemsGenerator = (items, acc = activeItemIds) =>
-        _.reduce(
-          items,
-          (acc, item) => {
-            if (item['expanded'] && acc.indexOf(item['id']) === -1) {
-              acc.push(item['id']);
-            }
+export type TreeStylesProps = never;
+
+const Tree: React.FC<WithAsProp<TreeProps>> &
+  FluentComponentStaticProps<TreeProps> & {
+    Item: typeof TreeItem;
+    Title: typeof TreeTitle;
+  } = props => {
+  const context: ProviderContextPrepared = React.useContext(ThemeContext);
+  const { setStart, setEnd } = useTelemetry(Tree.displayName, context.telemetry);
+  setStart();
+
+  const {
+    exclusive,
+    items,
+    renderItemTitle,
+    selectable,
+    children,
+    renderedItems,
+    className,
+    design,
+    styles,
+    variables,
+  } = props;
+
+  const ElementType = getElementType(props);
+  const unhandledProps = useUnhandledProps(Tree.handledProps, props);
+
+  const expandedItemsGenerator = (items, acc = []) =>
+    _.reduce(
+      items,
+      (acc, item) => {
+        if (item['expanded'] && acc.indexOf(item['id']) === -1) {
+          acc.push(item['id']);
+        }
 
-            if (item['items']) {
-              return expandedItemsGenerator(item['items'], acc);
-            }
+        if (item['items']) {
+          return expandedItemsGenerator(item['items'], acc);
+        }
 
-            return acc;
-          },
-          acc,
-        );
+        return acc;
+      },
+      acc,
+    );
 
-      expandedItemsGenerator(items);
-    }
+  const iterateItems = (items, acc = []) =>
```

Comment:
Can we move out `iterateItems` and `expandedItemsGenerator` of render scope? I.e. place outside component?

Reply:
Done